Add aarch64-darwin to flake.nix

ch1bo opened the pull request:
:snowflake: Enable builds of packages and devShells for `aarch64-darwin` and address #673.

:snowflake: Remove all the `std`/`tullia` stuff for `cicero` as we are not using it anymore.

:snowflake: Create a `hydraJobs` attribute on `flake.nix` and have it built on an [aarch64 CI](https://ci.zw3rk.com/jobset/input-output-hk-hydra/master#tabs-errors)

:snowflake: Adds http://cache.zw3rk.com/ as blessed cache to the repository

ch1bo committed:
Add aarch64-darwin to flake.nix

This should enable builds and devShells for aarch64-darwin.

ghost commented:
Trying to `nix develop` on a aarch64 machine yields:
```
% nix develop
warning: ignoring untrusted substituter 'https://cache.iog.io/'
warning: ignoring untrusted substituter 'https://cache.iog.io/'
warning: ignoring untrusted substituter 'https://hydra-node.cachix.org/'
trace: haskell-nix.haskellLib.cleanGit: /nix/store/l700g7agvs17j731lz6v328agrfqkp2k-source does not seem to be a git repository,
assuming it is a clean checkout.
error: builder for '/nix/store/2jsani4z3xqbs9cx7yd6v3qw109w787k-ghc-8.10.7.drv' failed with exit code 2;
       last 10 log lines:
       > xattr: [Errno 13] Permission denied: './utils/hsc2hs/dist-install/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/iserv/stage2_dyn/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/iserv/stage2_p/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/iserv/stage2/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/unlit/dist-install/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/runghc/dist-install/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './ghc/stage2/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './compiler/stage2/build/libiconv.dylib'
       > make[1]: *** [ghc.mk:833: install_darwin] Error 1
       > make: *** [Makefile:51: install] Error 2
       For full logs, run 'nix log /nix/store/2jsani4z3xqbs9cx7yd6v3qw109w787k-ghc-8.10.7.drv'.
error: 1 dependencies of derivation '/nix/store/j910ad1nrkxar9xc7rkrhhxgc527mxrq-ghc-8.10.7-configured-src.drv' failed to build
```

ghost commented:
I have tried the following but I am still facing the same permissions issue:
* Reboot the machine to ensure daemon is started and picks up new configuration
* uninstall and reinstall nix from scratch
Perhaps @angerman  will have some wisdom to share?

Note that single-user installation of nix is strongly discouraged on Mac OS

ch1bo committed:
Add hydraJobs to our flake

ch1bo committed:
Remove std/tullia

ch1bo committed:
Rename docker derivations to make flake well-formed

ch1bo commented:
Added `hydraJobs` and several jobs succesfully build on `aarch64-darwin` https://ci.zw3rk.com/jobset/input-output-hk-hydra/pullrequest-674#tabs-jobs

However, the static executables and the `hydra-cluster` for some of the shell environments makes problems.

It's also surprising that all the systems end up in the `hydraJobs.packages.hydra-node-static` for example. (This might be the reason why that derivation can't build an a single `system` host.)

ch1bo committed:
Only build hydra-node, hydra-tui and hydraw in CI

DevShells don't build for aarch64-darwin because the version of
cardano-node we use does not support it (yet).

The `...-static` static binaries also do only work for x86_64-linux.

The `docker-...` images do not build on hydra CI for a currently unknown
reason.

ch1bo committed:
Add hydra CI cache to get some aarch64-darwin cache hits

ch1bo commented:
FWIW this is the CI build result of this branch: https://ci.zw3rk.com/eval/27118

@abailly-iohk The `nix build .#hydra-node` should at least for you on `aarch64-darwin` especially after you enable the cache. Could you give it another go?

ghost commented:
Does not look promising:
```
[1/1/625 built, 0.0 MiB DL] building ghc-8.10.7 (unpackPhase): unpacking source archive /nix/store/c0ppmqaiiml15dgnww9zkh10iynn0xbr-ghc-8.10.7-aarch64-apple-darwin.tar.xz
```

ghost commented:
Here is my `/etc/nix/nix.conf` file:
```
build-users-group = nixbld
substituters = https://cache.nixos.org https://iohk.cachix.org https://hydra.iohk.io https://cache.iog.io
trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= iohk.cachix.org-1:DpRUyj7h7V830dp/i6Nti+NEO2/nhblbov/8MW7Rqoo= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
experimental-features = nix-command flakes
```

ghost commented:
More error when letting nix continue:
```
% nix build .#hydra-node
warning: ignoring untrusted substituter 'https://hydra-node.cachix.org'
warning: ignoring untrusted substituter 'https://cache.zw3rk.com'
trace: haskell-nix.haskellLib.cleanGit: /nix/store/2hr8hhy9mb0w4jck9kkl20s64fdi0v9m-source does not seem to be a git repository,
assuming it is a clean checkout.
error: builder for '/nix/store/2jsani4z3xqbs9cx7yd6v3qw109w787k-ghc-8.10.7.drv' failed with exit code 2;
       last 10 log lines:
       > xattr: [Errno 13] Permission denied: './utils/hsc2hs/dist-install/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/iserv/stage2_dyn/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/iserv/stage2_p/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/iserv/stage2/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/unlit/dist-install/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './utils/runghc/dist-install/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './ghc/stage2/build/tmp/libiconv.dylib'
       > xattr: [Errno 13] Permission denied: './compiler/stage2/build/libiconv.dylib'
       > make[1]: *** [ghc.mk:833: install_darwin] Error 1
       > make: *** [Makefile:51: install] Error 2
       For full logs, run 'nix log /nix/store/2jsani4z3xqbs9cx7yd6v3qw109w787k-ghc-8.10.7.drv'.
```

ch1bo commented:
@abailly-iohk can you make sure you enable the https://cache.zw3rk.com substituter? I see still `warning: ignoring untrusted substituter 'https://cache.zw3rk.com'` in your output. You should have been asked by `nix` whether you want to trust this cache or not. You can use `accept-flake-config = true` in your `nix.conf` if you want to always trust the ones listed in a `flake.nix` (there should also be a way to enable it only for one invocation).

ghost commented:
I of course said `y` to all nix's questions but it does not seem to reflect it in the settings 🤷

ghost commented:
I have added the suggested configuration to my `/etc/nix/nix.conf` and _restarted_ the nix daemon but it did not change anything. I also tried remove `~/.cache/nix` and `~/.local/share/nix`

angerman commented:
Ahhhhhhhhhh... how can I reproduce @abailly-iohk's behavior?

abailly commented:
@angerman We are going to troubleshoot my settings with @ch1bo, I am probably doing something blatantly wrong for a nix expert

angerman commented:
> @angerman We are going to troubleshoot my settings with @ch1bo, I am probably doing something blatantly wrong for a nix expert

No. You did not. You did something innocent at best. We need to figure out how. No one should need to be a nix expert. This should "just work" if it's not. We need to figure out why, and how to prevent this from happening in the future.

Commit:
Added note for Mac OS X users
